lib/iris/fileformats/netcdf.py

```diff
@@ -1924,36 +1946,58 @@ def set_packing_ncattrs(cfvar):
             self._dataset.file_format in ('NETCDF3_CLASSIC',
                                           'NETCDF3_64BIT')):
 
-            if packing is None:
-                # Determine whether there is a cube MDI value.
-                fill_value = get_fill_value(cube.core_data())
-
             # Get the values in a form which is valid for the file format.
             data = self._ensure_valid_dtype(cube.data, 'cube', cube)
 
-            if packing is None:
-                dtype = data.dtype.newbyteorder('=')
-
-            # Create the cube CF-netCDF data variable with data payload.
-            cf_var = self._dataset.createVariable(
-                cf_name, dtype, dimension_names,
-                fill_value=fill_value, **kwargs)
-            set_packing_ncattrs(cf_var)
-            cf_var[:] = data
-
+            def store(data, cf_var, fill_value):
+                cf_var[:] = data
+                is_masked = ma.isMaskedArray(data)
```

Comment:
Here too should be using "ma.is_masked" instead of "ma.isMaskedArray".
As stated [here](https://github.com/SciTools/iris/issues/2748#issuecomment-324578818) "make no difference between masked + non-masked arrays"